javaslang/src/test/java/javaslang/control/OptionTest.java

```diff
@@ -190,6 +190,30 @@ public void shouldReturnAlternativeOnOrElseSupplierIfValueIsNotDefined() {
         assertThat(Option.none().orElse(() -> opt)).isSameAs(opt);
     }
 
+    @Test
+    public void shouldReturnSelfOnOrElseOfIfValueIsPresent() {
```

Comment:
Could you please provide a more realistic use case?
A convincing real-world example, where this new functionality is useful (i.e. no dummy values).
(I don't decide anything, I'm just curious. @danieldietrich is the boss :smiley:)